cloud-consumption/src/test/java/com/sequenceiq/consumption/job/storage/StorageConsumptionJobTest.java

```diff
@@ -0,0 +1,104 @@
+package com.sequenceiq.consumption.job.storage;
+
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGenerator;
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareInternalCrnGeneratorFactory;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.consumption.domain.Consumption;
+import com.sequenceiq.consumption.flow.ConsumptionReactorFlowManager;
+import com.sequenceiq.consumption.service.ConsumptionService;
+
+import io.opentracing.Tracer;
+
+@ExtendWith(MockitoExtension.class)
+public class StorageConsumptionJobTest {
+
+    private static final Long LOCAL_ID = 42L;
+
+    private static final String REMOTE_CRN = "remote-crn";
+
+    private static final Long CONSUMPTION_ID = 1L;
+
+    @Mock
+    private ConsumptionService consumptionService;
+
+    @Mock
+    private ConsumptionReactorFlowManager flowManager;
+
+    @Mock
+    private RegionAwareInternalCrnGeneratorFactory regionAwareInternalCrnGeneratorFactory;
+
+    @Mock
+    private Consumption consumption;
+
+    @Mock
+    private RegionAwareInternalCrnGenerator crnGenerator;
+
+    @Mock
+    private JobExecutionContext jobExecutionContext;
+
+    @InjectMocks
+    private StorageConsumptionJob underTest;
+
+    @BeforeEach
+    public void setUp() {
+        Tracer tracer = Mockito.mock(Tracer.class);
+        underTest = new StorageConsumptionJob(tracer);
+        MockitoAnnotations.openMocks(this);
```

Comment:
Is it needed?

Reply:
I tried it a few ways, but it seems like even with @InjectMocks the @Mock-ed resources do not get injected into underTest, but if openMocks is called then they are injected. Checking the other XYJobTests it seems like they similarly call openMocks.

Reply:
I suspect that is because of the non-default constructor `StorageConsumptionJob(Tracer tracer)`. Mockito will pick this up and try injecting things via constructor only, hence missing the `@inject private` fields. In this specific case, there is no `io.opentracing.Tracer` mock present at all, so likely `null` will be passed to the above constructor. :smile: